misc/dbt-materialize/dbt/adapters/materialize/impl.py

```diff
@@ -69,3 +126,27 @@ def list_relations_without_caching(
             )
 
         return relations
+
+    def _get_cluster(self) -> str:
+        _, table = self.execute("SHOW CLUSTER", fetch=True)
+        if len(table) == 0 or len(table[0]) == 0:
+            raise RuntimeException("Could not get current cluster: no results")
+        return str(table[0][0])
+
+    def _set_cluster(self, cluster: str):
+        self.execute("SET CLUSTER = %s" % cluster)
+
+    def pre_model_hook(self, config: Mapping[str, Any]) -> Optional[str]:
+        default_cluster = self.config.credentials.cluster
+        cluster = config.get("cluster", default_cluster)
+        if cluster == default_cluster or cluster is None:
+            return None
+        previous = self._get_cluster()
+        self._set_cluster(cluster)
```

Comment:
Tried to explain this on Slack, but I thought it might be easier as a line comment! It's specifically _this_ adjustment of the `cluster` variable that worries me. If an exception is thrown at the wrong moment, I think it could result in the connection not having its `cluster` reset to the original value, and dbt continuing to build some other models with the wrong cluster active!

Is it possible to do what you do for indexes and change the [`materialize__create_materialized_view_as` macro](https://github.com/MaterializeInc/materialize/blob/47913de4350105849ce3ef979823c5b59c1442fe/misc/dbt-materialize/dbt/include/materialize/macros/adapters.sql#L28) to have a block like:

```
{% if cluster %}
in cluster {{ cluster }}
{% endif %}
```

You'll have to pass in the cluster as an argument to the macro, but it seems like it should be available as `config.get('cluster')` when you invoke the macro here? 

https://github.com/MaterializeInc/materialize/blob/4c8fc95431a7807f69b44a7ef7dceef975467b4f/misc/dbt-materialize/dbt/include/materialize/macros/materializations/materialized_view.sql#L35

Reply:
Thank you for pointing this out! Updated the create_view and create_materialized_view functions to handle.